tools/topology/CMakeLists.txt

```diff
@@ -47,11 +47,11 @@ macro(add_alsatplg2_command conf_header conf_target input_name output_name inclu
 		DEPENDS ${conf_header} ${conf_target}
 		OUTPUT ${output_name}.tplg
 
-		COMMAND cat ${conf_header} ${input_name}.conf > ${output_name}.conf
+		COMMAND cat ${conf_header} ${input_name}.conf > ../${output_dir}/${output_name}.conf
 
 		# -p to pre-process Topology2.0 conf file
 		COMMAND ALSA_CONFIG_DIR=${CMAKE_SOURCE_DIR}/topology/topology2 alsatplg \$\${VERBOSE:+-v 1}
-		        -I ${include_path} -D "'${defines}'" -p -c ${output_name}.conf -o ${output_name}.tplg
+		        -I ${include_path} -D "'${defines}'" -p -c ../${output_dir}/${output_name}.conf -o ../${output_dir}/${output_name}.tplg
```

Comment:
>  I'm afraid this could confuse not just people but even surprise CMake itself.

For sure it will right now: there is a mismatch with the OUTPUT value.